server/plugin/virtual_agent.go

```diff
@@ -457,13 +457,21 @@ func (p *Plugin) getPostActionOptions(options []Option) []*model.PostActionOptio
 	return postOptions
 }
 
-func (p *Plugin) CreateMessageAttachment(fileID string) (*MessageAttachment, error) {
+func (p *Plugin) CreateMessageAttachment(fileID, userID string) (*MessageAttachment, error) {
 	var attachment *MessageAttachment
 	fileInfo, appErr := p.API.GetFileInfo(fileID)
 	if appErr != nil {
 		return nil, fmt.Errorf("error getting the file info. Error: %s", appErr.Message)
 	}
 
+	if fileInfo.DeleteAt != 0 {
+		return nil, fmt.Errorf("file is deleted from the server")
+	}
+
+	if fileInfo.CreatorId != userID {
```

Comment:
Should we instead be checking `p.API.HasPermissionToChannel(userID, fileInfo.ChannelId, model.PermissionReadChannel)`?